x/fee_grant/internal/ante/fee.go

```diff
@@ -0,0 +1,94 @@
+package ante
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	// we depend on the auth module internals... maybe some more of this can be exported?
+	// but things like `x/auth/types/FeeCollectorName` are quite clearly tied to it
+	authante "github.com/cosmos/cosmos-sdk/x/auth/ante"
+	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/cosmos/cosmos-sdk/x/fee_grant/internal/keeper"
+	"github.com/cosmos/cosmos-sdk/x/fee_grant/internal/types/tx"
+
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+var (
+	_ GrantedFeeTx = (*tx.FeeGrantTx)(nil) // assert FeeGrantTx implements GrantedFeeTx
+)
+
+// GrantedFeeTx defines the interface to be implemented by Tx to use the GrantedFeeDecorator
+type GrantedFeeTx interface {
+	sdk.Tx
+	GetGas() uint64
+	GetFee() sdk.Coins
+	FeePayer() sdk.AccAddress
+	MainSigner() sdk.AccAddress
+}
+
+// DeductGrantedFeeDecorator deducts fees from the first signer of the tx
+// If the first signer does not have the funds to pay for the fees, return with InsufficientFunds error
+// Call next AnteHandler if fees successfully deducted
+// CONTRACT: Tx must implement GrantedFeeTx interface to use DeductGrantedFeeDecorator
+type DeductGrantedFeeDecorator struct {
+	ak authkeeper.AccountKeeper
+	dk keeper.Keeper
+	sk authtypes.SupplyKeeper
```

Comment:
not sure why do are we using the `SupplyKeeper` interface exposed to `auth`. This module should declare it's own expected keeper interface

Reply:
will do